sn_networking/src/event.rs

```diff
@@ -425,7 +428,11 @@ impl SwarmDriver {
                     .kademlia
                     .remove_peer(&failed_peer_id)
                 {
-                    self.send_event(NetworkEvent::PeerRemoved(*dead_peer.node.key.preimage()));
+                    self.connected_peers -= 1;
```

Comment:
for safty, better using `saturating_sub`
same using `saturating_add` if possible

Reply:
Thank you for the suggestion. Updated the PR